src/app/CommandHandler.h

```diff
@@ -158,6 +160,14 @@ class CommandHandler : public Messaging::ExchangeDelegate
      */
     CommandHandler(Callback * apCallback);
 
+    /*
+     * Constructor to override number of supported paths per invoke.
+     *
+     * The callback and command path registry passed in has to outlive this CommandHandler object.
+     * Created for testing purposes.
+     */
+    CommandHandler(Callback * apCallback, CommandPathRegistry * mCommandPathRegistry);
```

Comment:
Suggest making a new class "TestOnlyMarker" and you have to pass a constant instance of that class as argument (`kThisIsForTestOnly`), so that it's clear that you have to go out of your way to make the construction site be test-only.